lsst/ts/laser/csc.py

```diff
@@ -75,36 +88,56 @@ def __init__(self, address, configuration, frequency=1, initial_state=State.STAN
         None
 
         """
-        while True:
-            try:
-                self.model.publish()
-            except TimeoutError as te:
-                self.fault()
-            if self.model._laser.CPU8000.power_register.register_value == "FAULT" or \
-                    self.model._laser.M_CPU800.power_register.register_value == "FAULT" or \
-                    self.model._laser.M_CPU800.power_register_2.register_value == "FAULT":
+        try:
+            while True:
+                if self.summary_state is not salobj.State.FAULT:
+                    try:
+                        self.model.publish()
+                    except TimeoutError as te:
+                        self.fault()
+                        self.evt_errorCode.set_put(
+                            errorCode=LaserErrorCode.timeout_error,
+                            errorReport=te.msg,
+                            traceback=traceback.format_exc())
+                    if self.model._laser.CPU8000.power_register.register_value == "FAULT" or \
+                        self.model._laser.M_CPU800.power_register.register_value == "FAULT" or \
+                        self.model._laser.M_CPU800.power_register_2.register_value == "FAULT" \
+                            and self.summary_state is not salobj.State.FAULT:
+                                self.fault()
+                                self.evt_errorCode.set_put(
+                                    errorCode=LaserErrorCode.hw_cpu_error,
+                                    errorReport=f"Code:{self._laser.CPU8000.fault_register.fault}",
```

Comment:
you are only broadcasting `CPU8000` error registry but you check two other registries on `M_CPU800`. Is there a reason for that? Should you report all of it?